### PR TITLE
Simplify in-progress state when submitting User Input answers.

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #6323 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by hussain-t — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. The changes LGTM. However, I did a bit of testing and found the `What are your top goals for this site?` section’s checkboxes are editable (not disabled). Ideally, I think the `Edit` link should be disabled, and we shouldn’t let the users edit while submitting. WDYT?


https://user-images.githubusercontent.com/24408261/211786538-71aa8126-ba12-4199-84bf-9811908b6201.mov



cc: @aaemnnosttv

**Comment by sashadoes** [comment]
@hussain-t thanks for the review and feedback. I made an update.

**Review by hussain-t — APPROVED** [review]
Thanks, @sashadoes. LGTM 👍

File: assets/js/components/user-input/UserInputPreview.js [hunk 1]
```diff
@@ -175,18 +185,25 @@ export default function UserInputPreview( props ) {
 							<Button
 								className="googlesitekit-user-input__buttons--next"
 								onClick={ onSaveClick }
+								disabled={ isScreenLoading }
 							>
 								{ __( 'Save', 'google-site-kit' ) }
+								{ isScreenLoading && (
+									<Spinner isSaving={ isScreenLoading } />
```
Comment (eugene-manuilov) [inline assets/js/components/user-input/UserInputPreview.js:192]: We have the `SpinnerButton` component for this purposes. Could you please update this component to use it instead?

File: assets/js/components/user-input/CancelUserInputButton.js [hunk 2]
```diff
@@ -30,7 +35,7 @@ import { CORE_LOCATION } from '../../googlesitekit/datastore/location/constants'
 import Link from '../Link';
 const { useSelect, useDispatch } = Data;
 
-export default function CancelUserInputButton() {
+export default function CancelUserInputButton( { disabled = false } ) {
```
Comment (eugene-manuilov) [inline assets/js/components/user-input/CancelUserInputButton.js:38]: The default state for the `disabled` property is `undefined`. Let's stick with it:
```suggestion
export default function CancelUserInputButton( { disabled } ) {
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @sashadoes. Mostly looks good to me. Added two comments for you. Please, take a look.

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Actually, just noticed that this PR has merge conflicts. @sashadoes, could you please resolve all merge conflicts?

**Review by eugene-manuilov — APPROVED** [review]
Looks good now

**Pull request merged**